Assets/MRTK/Tests/PlayModeTests/GridObjectCollectionTests.cs

```diff
@@ -93,40 +93,139 @@ public IEnumerator TestAnchors()
             }
             yield return null;
         }
+        
+        /// <summary>
+        /// Tests that grid lays out object correctly for all different alignment options
+        /// </summary>
+        [UnityTest]
+        public IEnumerator TestGridAlignment()
+        {
+            // Create a grid
+            GameObject go = new GameObject();
+            go.name = "grid";
+            var grid = go.AddComponent<GridObjectCollection>();
+            grid.Distance = 0.75f;
+            grid.CellWidth = 0.15f;
+            grid.CellHeight = 0.15f;
+
+            grid.Layout = LayoutOrder.ColumnThenRow;
+            grid.Columns = 2;
+
+            grid.Anchor = LayoutAnchor.UpperCenter;
+
+            for (int i = 0; i < 3; i++)
+            {
+                var child = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                child.transform.parent = go.transform;
+                child.transform.localScale = Vector3.one * 0.1f;
+            }
+
+            grid.ColAlignment = LayoutHorizontalAlignment.Left;
+            grid.UpdateCollection();
+
+            int expectedIdx = 0;
+            foreach (LayoutHorizontalAlignment alignment in Enum.GetValues(typeof(LayoutHorizontalAlignment)))
+            {
+                grid.ColAlignment = alignment;
+                grid.UpdateCollection();
+                int j = 0;
+                foreach (Transform child2 in go.transform)
```

Comment:
Don't use numbered variable names (i.e. child2 -> childTransform).

It's odd in this case to see two things named child, but one is a transform and another is a primitive (one of the weird parts about having numerically named things)